staging/src/k8s.io/apiextensions-apiserver/pkg/controller/status/naming_controller.go

```diff
@@ -100,11 +101,15 @@ func (c *NamingConditionController) getAcceptedNamesForGroup(group string) (allR
 			continue
 		}
 
+		if curr.GetClusterName() != clusterName {
+			continue
+		}
+
 		// for each item here, see if we have a mutation cache entry that is more recent
 		// this makes sure that if we tight loop on update and run, our mutation cache will show
 		// us the version of the objects we just updated to.
 		item := curr
-		obj, exists, err := c.crdMutationCache.GetByKey(curr.Name)
+		obj, exists, err := c.crdMutationCache.GetByKey(genericapirequest.ToClusterAwareKey(curr.GetClusterName(), curr.Name))
```

Comment:
ToClusterAwareKey would really need to be in cache in the long run - it's a client problem, not a server problem (genericapirequest is a server package).  You might want to create a new package in client-go to hold this.

Reply:
Yes, I was already wondering about creating a new package in `client-go` for this. Seems I just got the confirmation now :-)
I'll do it.